components/search/src/selector.rs

```diff
@@ -1,27 +1,65 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-use crate::{error::Error, SearchApiResult};
+
+use crate::{error::Error, RefinedConfig, SearchApiResult, SearchUserEnvironment};
 use error_support::handle_error;
 
-pub struct SearchEngineSelector();
+/// SearchEngineSelector parses the JSON configuration for
+/// search engines and returns the applicable engines depending
+/// on their region + locale.
+#[derive(Default, uniffi::Object)]
+pub struct SearchEngineSelector {}
 
+#[uniffi::export]
 impl SearchEngineSelector {
+    #[uniffi::constructor]
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Sets the search configuration from the given string. This allows for
+    /// reprocessing of the configuration if it has not changed since the
+    /// previous update, e.g. to optimise test running.
+    #[handle_error(Error)]
+    pub fn set_search_config(&self, _configuration: String) -> SearchApiResult<()> {
+        Err(Error::NotImplemented)
+    }
+
+    /// Clears the search configuration from memory if it is known that it is
+    /// not required for a time.
+    pub fn clear_search_config(&self) {}
```

Comment:
How or when does it know that it is not required? What's the conditions for that?